src/slurmdbd_ops.py

```
@@ -0,0 +1,447 @@
+    def __init__(self):
+        """Set the initial attribute values."""
+        self._slurm_component = "slurmdbd"
+
+        self._slurm_state_dir = Path("/var/spool/slurmdbd")
+        self._jwt_rsa_key_file = self._slurm_state_dir / "jwt_hs256.key"
+        self._munge_key_path = Path("/etc/munge/munge.key")
+        self._slurmdbd_conf_dir = Path("/etc/slurm")
+        self._slurmdbd_conf_file = self._slurmdbd_conf_dir / "slurmdbd.conf"
+        self._slurmdbd_log_dir = Path("/var/log/slurm")
+        self._slurmdbd_log_file = self._slurmdbd_log_dir / "slurmdbd.log"
+        self._slurm_user = "slurm"
+        self._slurm_group = "slurm"
+        self._user_group = f"{self._slurm_user}:{self._slurm_group}"
+
+        self._munge_package = CharmedHPCPackageLifecycleManager("munge")
+        self._slurmdbd_package = CharmedHPCPackageLifecycleManager("slurmdbd")
+
+    def write_slurmdbd_conf(self, slurmdbd_parameters: dict) -> None:
+        """Render slurmdbd.conf."""
+        slurmdbd_conf = self._slurmdbd_conf_file
+
+        if slurmdbd_conf.exists():
+            slurmdbd_conf.write_bytes(os.urandom(2048))
+            slurmdbd_conf.unlink()
+
+        header = textwrap.dedent(
+            f"""
+            #
+            # {self._slurmdbd_conf_file} generated at {datetime.now()}
+            #
+
+            """
+        )
+        slurmdbd_conf.write_text(
+            header + "\n".join([f"{k}={v}" for k, v in slurmdbd_parameters.items() if v != ""])
+        )
+
+        slurmdbd_conf.chmod(0o600)
+
+        try:
+            subprocess.call(["chown", self._user_group, f"{slurmdbd_conf}"])
+        except subprocess.CalledProcessError as e:
+            raise (e)
+
+    def get_munge_key_from_fs(self) -> str:
+        """Return the munge_key from the munge.key on the filesystem."""
+        munge_key = ""
+        if self._munge_key_path.exists():
+            munge_key = b64encode(self._munge_key_path.read_bytes()).decode()
+        return munge_key
```

Should we throw an error if there's no munge key? Or is it okay for this method to return an empty string?

---

removed this function as it is unused